### PR TITLE
Setup backend

### PR DESCRIPTION
**Pull request by cassiemak**
Scraped data and linked database to the frontend!!

### CODE REVIEW

File: app/assets/javascripts/angular/angular-config.js.erb [hunk 1]
```diff
@@ -3,25 +3,27 @@ var app = angular.module('fritter', ['ngRoute']);
 app.config(['$routeProvider',function($routeProvider){
   $routeProvider.when('/', {
     templateUrl: "<%= asset_path('pin-index.html') %>",
-    controller: 'HomeCtrl',
+    controller: 'HomeCtrl'
   })
 }])
 
 app.config(['$routeProvider',function($routeProvider){
```
Comment (paulld) [inline app/assets/javascripts/angular/angular-config.js.erb]: There's no need to repeat app.config all the time, neither $routeProvider, for each route. you can do something like:

```
app.config(['$routeProvider',function($routeProvider){
    $routeProvider
        .when('...', {   ...   })
        .when('...', {   ...   })
        .when('...', {   ...   })
        .when('...', {   ...   })
```

**Comment by cassiemak** [comment]
I just made an extra commit to link the individual pins. 
REALLY. all linked now... 

Next is to do is to put in the partial (for user info and for header)

File: app/assets/javascripts/angular/angular-config.js.erb (hunk 1 above)
Comment (cassiemak) [inline app/assets/javascripts/angular/angular-config.js.erb]: OK! Will revise

File: app/assets/stylesheets/pin-index.css.scss [hunk 2]
```diff
@@ -111,3 +113,12 @@ p.description{
   }
 }
 
+.pin-show-link{
```
Comment (paulld) [inline app/assets/stylesheets/pin-index.css.scss:116]: Nice one!

File: app/assets/stylesheets/pin-index.css.scss [hunk 3]
```diff
@@ -90,7 +92,7 @@ h2.category{
   margin-bottom: 10px;
   font-size: 10px;
   overflow-wrap: break-word;
-  color: $primary-background-color;
+  color: gray;
```
Comment (paulld) [inline app/assets/stylesheets/pin-index.css.scss:95]: Why did you change to a default gray?

Comment (cassiemak) [inline app/assets/stylesheets/pin-index.css.scss:95]: The default color is too light, thus change it to gray to show
but will have to work with Dale to work out color schemes and then put in the primary color opacity function later.

File: app/views/categories/show.json.jbuilder [hunk 4]
```diff
@@ -0,0 +1 @@
+json.name @category.name
```
Comment (paulld) [inline app/views/categories/show.json.jbuilder:1]: You should be consistent in the way you set up your json. You currently have:
- pins#index -> `{ pins: [ {id: 1, ...}, {id: 2, ...},... ] }`
- pins#show -> `{ pin: [ {id: 1, ...} ] }`
- categories#index -> `{ categories: [ {id: 1, ...}, {id: 2, ...},... ] }`
- categories#index -> `{id: 1, ...}`     (without "categories" or "category" at the top level)

---> I think the best practice (quoting [jsonapi.org](http://jsonapi.org/format/)) is to always have the same structure with the model name in plural as the key, that takes an array of 1 or more items, no matter how many instances of the model you are displaying. For pins#show, it would then be exactly similar as for index: `{ pins: [ {id: 1, ...} ] }`

**Comment by paulld** [comment]
I'm done with my comments, you can check and make changes when necessary!
